src/librustc/session/mod.rs

```diff
@@ -179,26 +179,15 @@ impl Session {
     pub fn track_errors<F, T>(&self, f: F) -> Result<T, usize>
         where F: FnOnce() -> T
     {
-        let mut count = self.err_count();
+        let count = self.err_count();
         let result = f();
-        count -= self.err_count();
+        let count = self.err_count() - count;
```

Comment:
Nit: can we rename these variables to

    let old_count = ...;
    let new_errors = ... - old_count;